pages/api/user/index.ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { ResponseUtil } from "@utils/responseUtil";
+import { getUsers } from "@database/user";
+
+/**
+ * TODO: Write documentation for function
+ * @param req
+ * @param res
+ * @returns
+ */
+export default async function handle(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    const responseUtil = new ResponseUtil();
+    switch (req.method) {
+        case "GET": {
+            const users = await getUsers();
+            responseUtil.returnSuccess(res, undefined, users);
+            res.status(200).json({});
+            break;
+        }
+        case "POST": {
+            // TODO:
+            break;
+        }
+        case "PUT": {
+            // TODO:
+            break;
+        }
+        case "DELETE": {
+            // TODO:
+            break;
+        }
+        default:
+            res.setHeader("ALLOW", ["GET", "POST", "PUT", "DELETE"]);
```

Comment:
allowed methods can be passed into util function

Reply:
Let's do that instead

Reply:
done